### PR TITLE
AVL tree in python

### PR DESCRIPTION
**Pull request by sakshikhachane**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented AVL tree in python. 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes : #2131

### CODE REVIEW

File: Python/ds/AVL_Tree.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
+"""
+AVL tree is a self-balancing Binary Search Tree (BST) where the difference between heights of left and right subtrees cannot be more than one for all nodes and if at any time they differ by more than one, rebalancing is done to restore this property. Each node maintains extra information called a balance factor whose value is either -1, 0 or +1 where balance factor of a node in an AVL tree is the difference between the height of the left subtree and that of the right subtree of that node.
```
Comment (atarax665) [inline Python/ds/AVL_Tree.py]: Justify align this docstring.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/ds/AVL_Tree.py (hunk 1 above)
Comment (sakshikhachane) [reply Python/ds/AVL_Tree.py]: @atarax665 Can you please tell me how to justify align docstring in the python file? I was unable to find a proper solution for this and thus brought the lines one below the other.

**Review by HarshCasper — APPROVED** [review]

**Comment by virajmane** [comment]
@HarshCasper Can I add code for Deleting from a BST in python under DWoC?

**Comment by HarshCasper** [comment]
> @HarshCasper Can I add code for Deleting from a BST in python under DWoC?

I rarely take a look at merged PRs. Please make an Issue or start a discussion thread for any communication.

**Pull request merged**